kv/kvserver/protectedts/ptstorage: TestStorage failed

kv/kvserver/protectedts/ptstorage.TestStorage [failed](https://tanzanite.cluster.engflow.com/invocations/default/b300b368-e0fb-4f3b-ab26-9e3f16e52fee?testReportRun=24&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXIvcHJvdGVjdGVkdHMvcHRzdG9yYWdlOnB0c3RvcmFnZV90ZXN0) on master @ [7f45cb61df77419ded37619f701993ee410f6951](https://github.com/cockroachdb/cockroach/commits/7f45cb61df77419ded37619f701993ee410f6951):

Fatal error:

```
panic: use of Span after Finish. Span: propagate-ungraceful-stop. Finish previously called at: <stack not captured. Set debugUseAfterFinish>

```
Stack: 

```
goroutine 84310 [running]:
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).detectUseAfterFinish(0x9e994c0?)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:182 +0x114
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).RecordingType(0xc006508808)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:442 +0x18
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).IsVerbose(...)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:450
github.com/cockroachdb/cockroach/pkg/util/log.getSpan({0x66bfe70?, 0xc00a4b3bc0?})
	github.com/cockroachdb/cockroach/pkg/util/log/trace.go:27 +0x34
github.com/cockroachdb/cockroach/pkg/util/log.logfDepthInternal({0x66bfe70, 0xc00a4b3bc0}, 0x2, 0x1, 0x0, 0xd0?, {0x5266775, 0xc}, {0x0, 0x0, ...})
	github.com/cockroachdb/cockroach/pkg/util/log/channels.go:104 +0x4b4
github.com/cockroachdb/cockroach/pkg/util/log.logfDepth(...)
	github.com/cockroachdb/cockroach/pkg/util/log/channels.go:39
github.com/cockroachdb/cockroach/pkg/util/log.Infof(...)
	github.com/cockroachdb/cockroach/bazel-out/k8-fastbuild/bin/pkg/util/log/log_channels_generated.go:277
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Quiesce.func1()
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:577 +0x47
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```

pkg/kv/kvserver/protectedts/ptstorage_test/pkg/kv/kvserver/protectedts/ptstorage/storage_test.go:414: (protectOp.run)
	NOTICE: .Clock() called via implicit interface ApplicationLayerInterface;
HINT: consider using .ApplicationLayer().Clock() instead.
TIP: consider replacing the test server initialization from:
    ts, ... := serverutils.StartServer(t, ...)
    defer ts.Stopper().Stop(...)
to:
    srv, ... := serverutils.StartServer(t, ...)
    defer srv.Stopper().Stop(...)
    ts := srv.ApplicationLayer()

See also: https://go.crdb.dev/p/testserver-and-cluster-virtualization
=== RUN   TestStorage/Protect_-_already_exists_withMetaTable
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestStorage_Protect_-_already_exists_withMetaTable3245300945
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:157: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    storage_test.go:528: -- test log scope end --
=== RUN   TestStorage/Protect_-_too_many_spans_withMetaTable/7
=== RUN   TestStorage/Protect_-_too_many_spans_withMetaTable/9
=== RUN   TestStorage/Protect_-_unlimited_spans_withDeprecatedSpans_withMetaTable/3
=== RUN   TestStorage/Protect_-_already_exists_withDeprecatedSpans_withMetaTable
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestStorage_Protect_-_already_exists_withDeprecatedSpans_withMetaTable750626957
    test_log_scope.go:81: use -show-logs to present logs inline
    storage_test.go:528: -- test log scope end --
=== RUN   TestStorage/Protect_-_already_exists_withDeprecatedSpans_withMetaTable/0

pkg/kv/kvserver/protectedts/ptstorage_test/pkg/kv/kvserver/protectedts/ptstorage/storage_test.go:414: (protectOp.run)
	NOTICE: .Clock() called via implicit interface ApplicationLayerInterface;
HINT: consider using .ApplicationLayer().Clock() instead.
TIP: consider replacing the test server initialization from:
    ts, ... := serverutils.StartServer(t, ...)
    defer ts.Stopper().Stop(...)
to:
    srv, ... := serverutils.StartServer(t, ...)
    defer srv.Stopper().Stop(...)
    ts := srv.ApplicationLayer()

See also: https://go.crdb.dev/p/testserver-and-cluster-virtualization
=== RUN   TestStorage/Protect_-_too_many_bytes_withMetaTable
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestStorage_Protect_-_too_many_bytes_withMetaTable4053240358
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:157: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    storage_test.go:528: -- test log scope end --
=== RUN   TestStorage/Protect_-_unlimited_spans_withMetaTable/3
=== RUN   TestStorage/Protect_-_zero_timestamp_withDeprecatedSpans
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestStorage_Protect_-_zero_timestamp_withDeprecatedSpans1259542631
    test_log_scope.go:81: use -show-logs to present logs inline
    storage_test.go:528: -- test log scope end --
=== RUN   TestStorage/Protect_-_unlimited_bytes/3
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>run=24</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStorage.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-36347

kv/kvserver/protectedts/ptstorage.TestStorage [failed](https://tanzanite.cluster.engflow.com/invocations/default/e0d516de-9e96-4a3b-8b9c-87d9fd892cfe?testReportRun=24&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXIvcHJvdGVjdGVkdHMvcHRzdG9yYWdlOnB0c3RvcmFnZV90ZXN0) on master @ [ebf92d22a7e4afe2423e681d49c615104b5f0098](https://github.com/cockroachdb/cockroach/commits/ebf92d22a7e4afe2423e681d49c615104b5f0098):

Fatal error:

```
panic: use of Span after Finish. Span: propagate-ungraceful-stop. Finish previously called at: <stack not captured. Set debugUseAfterFinish>

```
Stack: 

```
goroutine 95465 [running]:
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).detectUseAfterFinish(0x9f2ee40?)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:182 +0x114
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).RecordingType(0xc00323ec08)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:442 +0x18
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).IsVerbose(...)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:450
github.com/cockroachdb/cockroach/pkg/util/log.getSpan({0x6722d58?, 0xc006e56ed0?})
	github.com/cockroachdb/cockroach/pkg/util/log/trace.go:27 +0x34
github.com/cockroachdb/cockroach/pkg/util/log.logfDepthInternal({0x6722d58, 0xc006e56ed0}, 0x2, 0x1, 0x0, 0xd0?, {0x52b0008, 0xc}, {0x0, 0x0, ...})
	github.com/cockroachdb/cockroach/pkg/util/log/channels.go:104 +0x4b4
github.com/cockroachdb/cockroach/pkg/util/log.logfDepth(...)
	github.com/cockroachdb/cockroach/pkg/util/log/channels.go:39
github.com/cockroachdb/cockroach/pkg/util/log.Infof(...)
	github.com/cockroachdb/cockroach/bazel-out/k8-fastbuild/bin/pkg/util/log/log_channels_generated.go:277
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Quiesce.func1()
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:577 +0x47
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestStorage/Protect_-_unlimited_bytes/1
=== RUN   TestStorage/Protect_-_unlimited_bytes_withDeprecatedSpans/2
=== RUN   TestStorage/Release_withDeprecatedSpans/0

pkg/kv/kvserver/protectedts/ptstorage_test/pkg/kv/kvserver/protectedts/ptstorage/storage_test.go:414: (protectOp.run)
	NOTICE: .Clock() called via implicit interface ApplicationLayerInterface;
HINT: consider using .ApplicationLayer().Clock() instead.
TIP: consider replacing the test server initialization from:
    ts, ... := serverutils.StartServer(t, ...)
    defer ts.Stopper().Stop(...)
to:
    srv, ... := serverutils.StartServer(t, ...)
    defer srv.Stopper().Stop(...)
    ts := srv.ApplicationLayer()

See also: https://go.crdb.dev/p/testserver-and-cluster-virtualization
=== RUN   TestStorage/MarkVerified_withMetaTable/2
=== RUN   TestStorage/Protect_-_already_verified_withDeprecatedSpans/0

pkg/kv/kvserver/protectedts/ptstorage_test/pkg/kv/kvserver/protectedts/ptstorage/storage_test.go:123: (init.func2)
	NOTICE: .Clock() called via implicit interface ApplicationLayerInterface;
HINT: consider using .ApplicationLayer().Clock() instead.
TIP: consider replacing the test server initialization from:
    ts, ... := serverutils.StartServer(t, ...)
    defer ts.Stopper().Stop(...)
to:
    srv, ... := serverutils.StartServer(t, ...)
    defer srv.Stopper().Stop(...)
    ts := srv.ApplicationLayer()

See also: https://go.crdb.dev/p/testserver-and-cluster-virtualization
=== RUN   TestStorage/Protect_-_too_many_spans_withDeprecatedSpans_withMetaTable/1
=== RUN   TestStorage/Protect_-_too_many_spans_withDeprecatedSpans_withMetaTable/6
=== RUN   TestStorage/UpdateTimestamp_withMetaTable/0

pkg/kv/kvserver/protectedts/ptstorage_test/pkg/kv/kvserver/protectedts/ptstorage/storage_test.go:414: (protectOp.run)
	NOTICE: .Clock() called via implicit interface ApplicationLayerInterface;
HINT: consider using .ApplicationLayer().Clock() instead.
TIP: consider replacing the test server initialization from:
    ts, ... := serverutils.StartServer(t, ...)
    defer ts.Stopper().Stop(...)
to:
    srv, ... := serverutils.StartServer(t, ...)
    defer srv.Stopper().Stop(...)
    ts := srv.ApplicationLayer()

See also: https://go.crdb.dev/p/testserver-and-cluster-virtualization
=== RUN   TestStorage/MarkVerified_withDeprecatedSpans_withMetaTable/3
=== RUN   TestStorage/Protect_-_already_exists/2
=== RUN   TestStorage/Protect_-_unlimited_bytes/3
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>run=24</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStorage.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver/protectedts/ptstorage.TestStorage [failed](https://tanzanite.cluster.engflow.com/invocations/default/dcf0dee8-bfef-4967-82c7-a4d88a6e796a?testReportRun=12&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXIvcHJvdGVjdGVkdHMvcHRzdG9yYWdlOnB0c3RvcmFnZV90ZXN0) on master @ [1084d88d63187bf66a2ee078e2f7ffd89cdfefba](https://github.com/cockroachdb/cockroach/commits/1084d88d63187bf66a2ee078e2f7ffd89cdfefba):

Fatal error:

```
panic: use of Span after Finish. Span: propagate-ungraceful-stop. Finish previously called at: <stack not captured. Set debugUseAfterFinish>

```
Stack: 

```
goroutine 97751 [running]:
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).detectUseAfterFinish(0x9ff7d80?)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:182 +0x114
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).RecordingType(0xc007e34008)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:442 +0x18
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).IsVerbose(...)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:450
github.com/cockroachdb/cockroach/pkg/util/log.getSpan({0x67bea98?, 0xc007e01170?})
	github.com/cockroachdb/cockroach/pkg/util/log/trace.go:27 +0x34
github.com/cockroachdb/cockroach/pkg/util/log.logfDepthInternal({0x67bea98, 0xc007e01170}, 0x2, 0x1, 0x0, 0xd0?, {0x533628c, 0xc}, {0x0, 0x0, ...})
	github.com/cockroachdb/cockroach/pkg/util/log/channels.go:104 +0x4b4
github.com/cockroachdb/cockroach/pkg/util/log.logfDepth(...)
	github.com/cockroachdb/cockroach/pkg/util/log/channels.go:39
github.com/cockroachdb/cockroach/pkg/util/log.Infof(...)
	github.com/cockroachdb/cockroach/bazel-out/k8-fastbuild/bin/pkg/util/log/log_channels_generated.go:277
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Quiesce.func1()
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:577 +0x47
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
    defer ts.Stopper().Stop(...)
to:
    srv, ... := serverutils.StartServer(t, ...)
    defer srv.Stopper().Stop(...)
    ts := srv.ApplicationLayer()

See also: https://go.crdb.dev/p/testserver-and-cluster-virtualization
=== RUN   TestStorage/Protect_-_unlimited_bytes_withMetaTable
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestStorage_Protect_-_unlimited_bytes_withMetaTable1472393837
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:157: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    storage_test.go:528: -- test log scope end --
=== RUN   TestStorage/Release_withDeprecatedSpans_withMetaTable
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestStorage_Release_withDeprecatedSpans_withMetaTable814258757
    test_log_scope.go:81: use -show-logs to present logs inline
    storage_test.go:528: -- test log scope end --
=== RUN   TestStorage/UpdateTimestamp_withDeprecatedSpans_withMetaTable/1
=== RUN   TestStorage/MarkVerified_withMetaTable
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestStorage_MarkVerified_withMetaTable2741003891
    test_log_scope.go:81: use -show-logs to present logs inline
    storage_test.go:528: -- test log scope end --
=== RUN   TestStorage/Protect_-_too_many_spans_withDeprecatedSpans_withMetaTable/7
=== RUN   TestStorage/Protect_-_unlimited_bytes
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestStorage_Protect_-_unlimited_bytes70185639
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:157: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
=== RUN   TestStorage/Protect_-_unlimited_bytes_withDeprecatedSpans/3
=== RUN   TestStorage/Protect_-_zero_timestamp_withDeprecatedSpans_withMetaTable
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestStorage_Protect_-_zero_timestamp_withDeprecatedSpans_withMetaTable2555253571
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:157: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    storage_test.go:528: -- test log scope end --
=== RUN   TestStorage/Protect_-_unlimited_spans_withMetaTable/3
=== RUN   TestStorage/Protect_-_zero_timestamp_withDeprecatedSpans/0
=== RUN   TestStorage/MarkVerified_withDeprecatedSpans
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestStorage_MarkVerified_withDeprecatedSpans3103215511
    test_log_scope.go:81: use -show-logs to present logs inline
    storage_test.go:528: -- test log scope end --
=== RUN   TestStorage/Protect_-_already_verified_withDeprecatedSpans_withMetaTable
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestStorage_Protect_-_already_verified_withDeprecatedSpans_withMetaTable657668008
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:157: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    storage_test.go:528: -- test log scope end --
=== RUN   TestStorage/Protect_-_already_verified_withMetaTable
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestStorage_Protect_-_already_verified_withMetaTable4002087519
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:157: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    storage_test.go:528: -- test log scope end --
=== RUN   TestStorage/Protect_-_too_many_spans_withDeprecatedSpans_withMetaTable/9
=== RUN   TestStorage/Protect_-_unlimited_bytes/3
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>run=12</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStorage.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

kv/kvserver/protectedts/ptstorage.TestStorage [failed](https://tanzanite.cluster.engflow.com/invocations/default/93e0e2c6-1e1b-4480-b257-144ddc8ca767?testReportRun=29&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXIvcHJvdGVjdGVkdHMvcHRzdG9yYWdlOnB0c3RvcmFnZV90ZXN0) on master @ [7488e090daa588c4d7c0f828c8006bb9b13a90f6](https://github.com/cockroachdb/cockroach/commits/7488e090daa588c4d7c0f828c8006bb9b13a90f6):

Fatal error:

```
panic: use of Span after Finish. Span: propagate-ungraceful-stop. Finish previously called at: <stack not captured. Set debugUseAfterFinish>

```
Stack: 

```
goroutine 70299 [running]:
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).detectUseAfterFinish(0xa0279c0?)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:182 +0x114
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).RecordingType(0xc00639f008)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:442 +0x18
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).IsVerbose(...)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:450
github.com/cockroachdb/cockroach/pkg/util/log.getSpan({0x67dfad8?, 0xc0047d56e0?})
	github.com/cockroachdb/cockroach/pkg/util/log/trace.go:27 +0x34
github.com/cockroachdb/cockroach/pkg/util/log.logfDepthInternal({0x67dfad8, 0xc0047d56e0}, 0x2, 0x1, 0x0, 0xd0?, {0x534e271, 0xc}, {0x0, 0x0, ...})
	github.com/cockroachdb/cockroach/pkg/util/log/channels.go:104 +0x4b4
github.com/cockroachdb/cockroach/pkg/util/log.logfDepth(...)
	github.com/cockroachdb/cockroach/pkg/util/log/channels.go:39
github.com/cockroachdb/cockroach/pkg/util/log.Infof(...)
	github.com/cockroachdb/cockroach/bazel-out/k8-fastbuild/bin/pkg/util/log/log_channels_generated.go:277
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Quiesce.func1()
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:577 +0x47
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
See also: https://go.crdb.dev/p/testserver-and-cluster-virtualization
=== RUN   TestStorage/GetRecord_-_does_not_exist_withDeprecatedSpans_withMetaTable/0
=== RUN   TestStorage/Protect_-_already_exists_withDeprecatedSpans/0

pkg/kv/kvserver/protectedts/ptstorage_test/pkg/kv/kvserver/protectedts/ptstorage/storage_test.go:414: (protectOp.run)
	NOTICE: .Clock() called via implicit interface ApplicationLayerInterface;
HINT: consider using .ApplicationLayer().Clock() instead.
TIP: consider replacing the test server initialization from:
    ts, ... := serverutils.StartServer(t, ...)
    defer ts.Stopper().Stop(...)
to:
    srv, ... := serverutils.StartServer(t, ...)
    defer srv.Stopper().Stop(...)
    ts := srv.ApplicationLayer()

See also: https://go.crdb.dev/p/testserver-and-cluster-virtualization
=== RUN   TestStorage/Protect_-_already_verified_withDeprecatedSpans
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestStorage_Protect_-_already_verified_withDeprecatedSpans180157230
    test_log_scope.go:81: use -show-logs to present logs inline
    storage_test.go:528: -- test log scope end --
=== RUN   TestStorage/Protect_-_simple_positive_withDeprecatedSpans
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestStorage_Protect_-_simple_positive_withDeprecatedSpans1048857393
    test_log_scope.go:81: use -show-logs to present logs inline
    storage_test.go:528: -- test log scope end --
=== RUN   TestStorage/Protect_-_unlimited_bytes_withDeprecatedSpans
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestStorage_Protect_-_unlimited_bytes_withDeprecatedSpans4217594741
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:157: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    storage_test.go:528: -- test log scope end --
=== RUN   TestStorage/Protect_-_unlimited_bytes_withDeprecatedSpans_withMetaTable
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestStorage_Protect_-_unlimited_bytes_withDeprecatedSpans_withMetaTable2316212519
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:157: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    storage_test.go:528: -- test log scope end --
=== RUN   TestStorage/Protect_-_already_verified_withDeprecatedSpans_withMetaTable/0

pkg/kv/kvserver/protectedts/ptstorage_test/pkg/kv/kvserver/protectedts/ptstorage/storage_test.go:123: (init.func2)
	NOTICE: .Clock() called via implicit interface ApplicationLayerInterface;
HINT: consider using .ApplicationLayer().Clock() instead.
TIP: consider replacing the test server initialization from:
    ts, ... := serverutils.StartServer(t, ...)
    defer ts.Stopper().Stop(...)
to:
    srv, ... := serverutils.StartServer(t, ...)
    defer srv.Stopper().Stop(...)
    ts := srv.ApplicationLayer()

See also: https://go.crdb.dev/p/testserver-and-cluster-virtualization
=== RUN   TestStorage/Protect_-_too_many_spans_withMetaTable/8
=== RUN   TestStorage/Protect_-_unlimited_bytes_withMetaTable/3
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>run=29</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStorage.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

kv/kvserver/protectedts/ptstorage.TestStorage [failed](https://tanzanite.cluster.engflow.com/invocations/default/f306225d-360a-4bbe-8ec8-cb228a97db16?testReportRun=10&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXIvcHJvdGVjdGVkdHMvcHRzdG9yYWdlOnB0c3RvcmFnZV90ZXN0) on master @ [3292e23e914f2b5a63e892fef1705ad476be4bd7](https://github.com/cockroachdb/cockroach/commits/3292e23e914f2b5a63e892fef1705ad476be4bd7):

Fatal error:

```
panic: use of Span after Finish. Span: propagate-ungraceful-stop. Finish previously called at: <stack not captured. Set debugUseAfterFinish>

```
Stack: 

```
goroutine 82029 [running]:
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).detectUseAfterFinish(0xa02ca80?)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:182 +0x114
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).RecordingType(0xc00a4cec08)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:442 +0x18
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).IsVerbose(...)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:450
github.com/cockroachdb/cockroach/pkg/util/log.getSpan({0x67e29b8?, 0xc00a9c1e30?})
	github.com/cockroachdb/cockroach/pkg/util/log/trace.go:27 +0x34
github.com/cockroachdb/cockroach/pkg/util/log.logfDepthInternal({0x67e29b8, 0xc00a9c1e30}, 0x2, 0x1, 0x0, 0xd0?, {0x5350bf9, 0xc}, {0x0, 0x0, ...})
	github.com/cockroachdb/cockroach/pkg/util/log/channels.go:104 +0x4b4
github.com/cockroachdb/cockroach/pkg/util/log.logfDepth(...)
	github.com/cockroachdb/cockroach/pkg/util/log/channels.go:39
github.com/cockroachdb/cockroach/pkg/util/log.Infof(...)
	github.com/cockroachdb/cockroach/bazel-out/k8-fastbuild/bin/pkg/util/log/log_channels_generated.go:277
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Quiesce.func1()
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:577 +0x47
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
	NOTICE: .Clock() called via implicit interface ApplicationLayerInterface;
HINT: consider using .ApplicationLayer().Clock() instead.
TIP: consider replacing the test server initialization from:
    ts, ... := serverutils.StartServer(t, ...)
    defer ts.Stopper().Stop(...)
to:
    srv, ... := serverutils.StartServer(t, ...)
    defer srv.Stopper().Stop(...)
    ts := srv.ApplicationLayer()

See also: https://go.crdb.dev/p/testserver-and-cluster-virtualization
=== RUN   TestStorage/Protect_-_too_many_spans_withDeprecatedSpans_withMetaTable/2
=== RUN   TestStorage/Protect_-_too_many_spans_withMetaTable/1
=== RUN   TestStorage/Protect_-_too_many_spans_withDeprecatedSpans_withMetaTable/1
=== RUN   TestStorage/Protect_-_too_many_spans_withMetaTable/4
=== RUN   TestStorage/Protect_-_unlimited_bytes_withDeprecatedSpans_withMetaTable/1
=== RUN   TestStorage/Protect_-_unlimited_bytes_withMetaTable
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestStorage_Protect_-_unlimited_bytes_withMetaTable4028139113
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:157: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
=== RUN   TestStorage/Release_withDeprecatedSpans/2
=== RUN   TestStorage/MarkVerified_withDeprecatedSpans_withMetaTable/3
=== RUN   TestStorage/Protect_-_no_targets_withMetaTable
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestStorage_Protect_-_no_targets_withMetaTable2554703773
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:157: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    storage_test.go:528: -- test log scope end --
=== RUN   TestStorage/Protect_-_too_many_spans_withDeprecatedSpans_withMetaTable
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestStorage_Protect_-_too_many_spans_withDeprecatedSpans_withMetaTable637580937
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:157: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    storage_test.go:528: -- test log scope end --
=== RUN   TestStorage/Release_withDeprecatedSpans_withMetaTable/2
=== RUN   TestStorage/UpdateTimestamp_--_does_not_exist_withDeprecatedSpans_withMetaTable
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestStorage_UpdateTimestamp_--_does_not_exist_withDeprecatedSpans_withMetaTable2121773356
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:157: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    storage_test.go:528: -- test log scope end --
=== RUN   TestStorage/UpdateTimestamp_withDeprecatedSpans
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestStorage_UpdateTimestamp_withDeprecatedSpans204039984
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:157: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    storage_test.go:528: -- test log scope end --
=== RUN   TestStorage/Protect_-_already_verified_withDeprecatedSpans
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestStorage_Protect_-_already_verified_withDeprecatedSpans1175140145
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:157: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    storage_test.go:528: -- test log scope end --
=== RUN   TestStorage/Protect_-_too_many_bytes_withMetaTable/3
=== RUN   TestStorage/Protect_-_unlimited_bytes_withMetaTable/3
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>run=10</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStorage.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

kv/kvserver/protectedts/ptstorage.TestStorage [failed](https://tanzanite.cluster.engflow.com/invocations/default/20281bff-5a92-4590-a88d-ef3bdc6b8a5a?testReportRun=25&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXIvcHJvdGVjdGVkdHMvcHRzdG9yYWdlOnB0c3RvcmFnZV90ZXN0) on master @ [ca56768b5244805e881d2f072dd8736931f6f3b8](https://github.com/cockroachdb/cockroach/commits/ca56768b5244805e881d2f072dd8736931f6f3b8):

Fatal error:

```
panic: use of Span after Finish. Span: propagate-ungraceful-stop. Finish previously called at: <stack not captured. Set debugUseAfterFinish>

```
Stack: 

```
goroutine 107976 [running]:
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).detectUseAfterFinish(0xa05b300?)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:182 +0x114
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).RecordingType(0xc00b7f6008)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:442 +0x18
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).IsVerbose(...)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:450
github.com/cockroachdb/cockroach/pkg/util/log.getSpan({0x680b7f8?, 0xc0090bb9b0?})
	github.com/cockroachdb/cockroach/pkg/util/log/trace.go:27 +0x34
github.com/cockroachdb/cockroach/pkg/util/log.logfDepthInternal({0x680b7f8, 0xc0090bb9b0}, 0x2, 0x1, 0x0, 0xd0?, {0x5366504, 0xc}, {0x0, 0x0, ...})
	github.com/cockroachdb/cockroach/pkg/util/log/channels.go:104 +0x4b4
github.com/cockroachdb/cockroach/pkg/util/log.logfDepth(...)
	github.com/cockroachdb/cockroach/pkg/util/log/channels.go:39
github.com/cockroachdb/cockroach/pkg/util/log.Infof(...)
	github.com/cockroachdb/cockroach/bazel-out/k8-fastbuild/bin/pkg/util/log/log_channels_generated.go:277
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Quiesce.func1()
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:577 +0x47
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:157: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    storage_test.go:528: -- test log scope end --
=== RUN   TestStorage/Protect_-_unlimited_spans_withMetaTable/1
=== RUN   TestStorage/Protect_-_zero_timestamp_withMetaTable/0
=== RUN   TestStorage/Protect_-_no_targets_withDeprecatedSpans_withMetaTable
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestStorage_Protect_-_no_targets_withDeprecatedSpans_withMetaTable4068719820
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:157: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    storage_test.go:528: -- test log scope end --
=== RUN   TestStorage/Protect_-_unlimited_bytes_withDeprecatedSpans/1
=== RUN   TestStorage/Protect_-_too_many_bytes_withDeprecatedSpans_withMetaTable
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestStorage_Protect_-_too_many_bytes_withDeprecatedSpans_withMetaTable4083113623
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:157: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    storage_test.go:528: -- test log scope end --
=== RUN   TestStorage/Protect_-_too_many_bytes_withMetaTable/3
=== RUN   TestStorage/Protect_-_unlimited_spans_withDeprecatedSpans_withMetaTable/3
=== RUN   TestStorage/Protect_-_already_verified/0

pkg/kv/kvserver/protectedts/ptstorage_test/pkg/kv/kvserver/protectedts/ptstorage/storage_test.go:123: (init.func2)
	NOTICE: .Clock() called via implicit interface ApplicationLayerInterface;
HINT: consider using .ApplicationLayer().Clock() instead.
TIP: consider replacing the test server initialization from:
    ts, ... := serverutils.StartServer(t, ...)
    defer ts.Stopper().Stop(...)
to:
    srv, ... := serverutils.StartServer(t, ...)
    defer srv.Stopper().Stop(...)
    ts := srv.ApplicationLayer()

See also: https://go.crdb.dev/p/testserver-and-cluster-virtualization
=== RUN   TestStorage/Protect_-_already_verified_withDeprecatedSpans_withMetaTable/0

pkg/kv/kvserver/protectedts/ptstorage_test/pkg/kv/kvserver/protectedts/ptstorage/storage_test.go:123: (init.func2)
	NOTICE: .Clock() called via implicit interface ApplicationLayerInterface;
HINT: consider using .ApplicationLayer().Clock() instead.
TIP: consider replacing the test server initialization from:
    ts, ... := serverutils.StartServer(t, ...)
    defer ts.Stopper().Stop(...)
to:
    srv, ... := serverutils.StartServer(t, ...)
    defer srv.Stopper().Stop(...)
    ts := srv.ApplicationLayer()

See also: https://go.crdb.dev/p/testserver-and-cluster-virtualization
=== RUN   TestStorage/Protect_-_too_many_spans_withDeprecatedSpans_withMetaTable/3
=== RUN   TestStorage/Protect_-_unlimited_spans_withDeprecatedSpans/1
=== RUN   TestStorage/Protect_-_too_many_spans_withMetaTable/6
=== RUN   TestStorage/Protect_-_unlimited_bytes/3
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>run=25</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #121335 kv/kvserver/protectedts/ptstorage: TestStorage failed [C-bug C-test-failure GA-blocker O-robot P-2 T-kv X-duplicate branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStorage.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/util/stop/BUILD.bazel
+++ b/pkg/util/stop/BUILD.bazel
@@ -12,6 +12,7 @@ go_library(
         "//pkg/util/log/logcrash",
         "//pkg/util/quotapool",
         "//pkg/util/syncutil",
+        "//pkg/util/timeutil",
         "//pkg/util/tracing",
         "@com_github_cockroachdb_errors//:errors",
     ],

--- a/pkg/util/stop/stopper.go
+++ b/pkg/util/stop/stopper.go
@@ -26,6 +26,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/util/log/logcrash"
 	"github.com/cockroachdb/cockroach/pkg/util/quotapool"
 	"github.com/cockroachdb/cockroach/pkg/util/syncutil"
+	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
 	"github.com/cockroachdb/cockroach/pkg/util/tracing"
 	"github.com/cockroachdb/errors"
 )
@@ -573,12 +574,6 @@ func (s *Stopper) IsStopped() <-chan struct{} {
 // Quiesce moves the stopper to state quiescing and waits until all
 // tasks complete. This is used from Stop() and unittests.
 func (s *Stopper) Quiesce(ctx context.Context) {
-	defer time.AfterFunc(5*time.Second, func() {
-		log.Infof(ctx, "quiescing...")
-	}).Stop()
-	defer time.AfterFunc(2*time.Minute, func() {
-		log.DumpStacks(ctx, "slow quiesce")
-	}).Stop()
 	defer s.recover(ctx)
 
 	func() {
@@ -600,7 +595,18 @@ func (s *Stopper) Quiesce(ctx context.Context) {
 		}
 	}()
 
+	start := timeutil.Now()
+	var loggedQuiescing, loggedSlowQuiescing bool
 	for s.NumTasks() > 0 {
+		since := timeutil.Since(start)
+		if !loggedQuiescing && since > 5*time.Second {
+			log.Infof(ctx, "quiescing...")
+			loggedQuiescing = true
+		}
+		if !loggedSlowQuiescing && since > 2*time.Minute {
+			log.DumpStacks(ctx, "slow quiesce")
+			loggedSlowQuiescing = true
+		}
 		time.Sleep(5 * time.Millisecond)
 	}
 }

